### PR TITLE
Offer details

### PR DESCRIPTION
**Pull request by AymanAlqoqa**
### Related:
related to #22

### Summary:
OfferDetails Page that contains all offer details
logic for application card:
- [x] logged in user is the offer owner:
  - if the applicant is not hired, show hire me button
  - if the application hire request is send, show 'hired request send' text.
  - if the applicant accepted the hire request, show 'accepted'
  - if the applicant refused the hire request, show 'refused'
- [x] logged in user is not offer owner:
  - if the applicant did not post his proposal yet, show input form to submit his proposal.
  - if the applicant posted his proposal, show his proposal and:
       - if the offer owner did not send hire request, show pending.
       - if the offer owner sent hire request, show accept or refuse buttons and the application status becomes accepted or refused.
      

### Notes:
Optional notes

### Checklist Criteria
- [ ] **Branch is up to date with master**
- [ ] **The project runs**
- - [ ] You can download and install locally, and run the project locally
- - [ ] All existing and new features run without bugs
- - [ ] The build script runs without error
- [ ] **Code is consistent**
- - [ ] Linter is used, no error messages
- - [ ] Code is consistent with agreed team style
- - [ ] Naming of files is consistent with project (e.g. all files named with camelCase)
- [ ] **No console logs**
- - [ ] Console.log is only used for crucial system messages (e.g. which port server is on).
- - [ ] Any console.logs used for debugging have been removed.

### CODE REVIEW

File: client/src/Components/OfferDetails/index.jsx [hunk 1]
```diff
@@ -1,6 +1,148 @@
...
+                      application={item}
+                      offerColor={this.offerColor}
+                    />
+                  </div>
+                ))}
+            </Col>
+          </div>
+        ) : (
+          // if logged in member is not offer owner
+          (!offer.status && (
```
Comment (ashatat) [inline client/src/Components/OfferDetails/index.jsx]: I think we agreed that the offer status default value is 'active'  so this won't work
I suggest you check if the logged in member have applied before hence `!memberApplication[0]`

File: client/src/Components/OfferDetails/SideCard/index.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import './style.css';
+
+const SideCard = props => {
+  const { title, items } = props;
```
Comment (ashatat) [inline client/src/Components/OfferDetails/SideCard/index.jsx]: validate props with propTypes

File: client/src/Components/OfferDetails/ApplicationCard/index.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    const { member_id: applicantMemberId } = this.props.application;
+  };
+
+  render() {
+    const {
+      full_name: fullName,
+      avatar,
+      discription,
+      status,
+    } = this.props.application;
```
Comment (ashatat) [inline client/src/Components/OfferDetails/ApplicationCard/index.jsx]: You need to validate using propTypes

File: client/src/Components/OfferDetails/ApplicationCard/index.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            </Card.Title>
+          </Card.Header>
+          <Card.Body>
+            <Card.Text>{discription}</Card.Text>
+            <div className="application-card__button__container">
+              <Button
+                className="application-card__button"
+                onClick={this.handleProfile}
+              >
+                {this.props.viewMyProfile ? 'View My Profile' : 'View Profile'}
```
Comment (ashatat) [inline client/src/Components/OfferDetails/ApplicationCard/index.jsx]: you can use destructuring props assignment

File: client/src/Components/OfferDetails/index.jsx [hunk 5]
```diff
@@ -1,6 +1,148 @@
...
+      fullName: 'Ayman AlQoqa',
+      username: 'Ayman321396',
+      avatar:
+        'https://m.media-amazon.com/images/M/MV5BMTcxOTk4NzkwOV5BMl5BanBnXkFtZTcwMDE3MTUzNA@@._V1_.jpg',
+    },
+  };
+
+  componentDidMount() {
+    const offerIdLength = window.location.href.split('/').length;
+    const offerId = window.location.href.split('/')[offerIdLength - 1];
```
[the remark this answers is not in the document]
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/index.jsx]: I think it is much better than my code
thanks

File: client/src/Components/OfferDetails/index.jsx [hunk 6]
```diff
@@ -1,6 +1,148 @@
...
+    // fetch offer details by offerId and save it in state
+    this.setState({ offer: offerData });
+    // fetch applications by offerId and save it in state
+    this.setState({ applications: applicationsData });
+    // take logged in userInfo from local storage and store it in state
+    // const userInfo = localStorage.getItem('userInfo');
+    // this.setState(userInfo);
```
[the remark this answers is not in the document]
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/index.jsx]: ok

File: client/src/Components/OfferDetails/index.jsx [hunk 7]
```diff
@@ -1,6 +1,148 @@
...
+
+  handleEndContract = () => {
+    // handle end contract login here....
+  };
+
+  render() {
+    const { offer } = this.state;
+    const { applications } = this.state;
+    const offerColor = this.offerColor(offer.status);
+    const { id: memberId } = this.state.userInfo;
```
[the remark this answers is not in the document]
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/index.jsx]: I think your are right
thanks

File: client/src/Components/OfferDetails/index.jsx (hunk 1 above)
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/index.jsx]: ok

File: client/src/Components/OfferDetails/SideCard/index.jsx (hunk 2 above)
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/SideCard/index.jsx]: ok

File: client/src/Components/OfferDetails/ApplicationCard/index.jsx (hunk 3 above)
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/ApplicationCard/index.jsx]: ok

File: client/src/Components/OfferDetails/ApplicationCard/index.jsx (hunk 4 above)
Comment (AymanAlqoqa) [reply client/src/Components/OfferDetails/ApplicationCard/index.jsx]: ok

**Pull request merged**